File: WebScrapingBackend/WebScraping/Entities/Yayin.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System;

namespace WebScraping.Entities
{
    public class Yayin
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }


        [BsonElement("Ad")]
        public string Ad { get; set; }


        [BsonElement("Yazarlar")]
        public string Yazarlar { get; set; }


        [BsonElement("Tur")]
        public string Tur { get; set; }


        [BsonElement("YayinlanmaTarihi")]
        public DateTime YayinlanmaTarihi { get; set; }


        [BsonElement("YayinciAdi")]
        public string YayinciAdi { get; set; }


        [BsonElement("AnahtarKelimelerAramaMotoru")]
        public string AnahtarKelimelerAramaMotoru { get; set; }


        [BsonElement("AnahtarKelimelerMakaleyeAit")]
        public string AnahtarKelimelerMakaleyeAit { get; set; }


        [BsonElement("Ozet")]
        public string Ozet { get; set; }


        [BsonElement("AlintiSayisi")]
        public int AlintiSayisi { get; set; }


        [BsonElement("DoiNumarasi")]
        public string DoiNumarasi { get; set; }


        [BsonElement("Url")]
        public string Url { get; set; }

        [BsonElement("PdfLinki")]
        public string PdfLinki { get; set; }
    }
}

File: webscrapingUI/src/app/common/yayin.ts
export class Yayin {
  id: string;
  ad: string;
  yazarlar: string;
  tur: string;
  yayinlanmaTarihi: string;
  yayinciAdi: string;
  anahtarKelimelerAramaMotoru: string;
  anahtarKelimelerMakaleyeAit: string;
  ozet: string;
  alintiSayisi: number;
  doiNumarasi: string;
  url: string;

  constructor(
      id: string,
      ad: string,
      yazarlar: string,
      tur: string,
      yayinlanmaTarihi: string,
      yayinciAdi: string,
      anahtarKelimelerAramaMotoru: string,
      anahtarKelimelerMakaleyeAit: string,
      ozet: string,
      alintiSayisi: number,
      doiNumarasi: string,
      url: string
  ) {
      this.id = id;
      this.ad = ad;
      this.yazarlar = yazarlar;
      this.tur = tur;
      this.yayinlanmaTarihi = yayinlanmaTarihi;
      this.yayinciAdi = yayinciAdi;
      this.anahtarKelimelerAramaMotoru = anahtarKelimelerAramaMotoru;
      this.anahtarKelimelerMakaleyeAit = anahtarKelimelerMakaleyeAit;
      this.ozet = ozet;
      this.alintiSayisi = alintiSayisi;
      this.doiNumarasi = doiNumarasi;
      this.url = url;
  }
}
